network/src/protocols/support_protocols.rs

```diff
@@ -0,0 +1,131 @@
+use crate::ProtocolId;
+use p2p::{
+    builder::MetaBuilder,
+    service::{BlockingFlag, ProtocolHandle, ProtocolMeta},
+    traits::ServiceProtocol,
+};
+use tokio_util::codec::length_delimited;
+
+pub enum SupportProtocols {
+    Ping,
+    Discovery,
+    Identify,
+    Feeler,
+    DisconnectMessage,
+    Sync,
+    Relay,
+    Time,
+    Alert,
+}
+
+impl SupportProtocols {
+    pub fn protocol_id(&self) -> ProtocolId {
+        match self {
+            SupportProtocols::Ping => 0,
+            SupportProtocols::Discovery => 1,
+            SupportProtocols::Identify => 2,
+            SupportProtocols::Feeler => 3,
+            SupportProtocols::DisconnectMessage => 4,
+            SupportProtocols::Sync => 100,
+            SupportProtocols::Relay => 101,
+            SupportProtocols::Time => 102,
+            SupportProtocols::Alert => 110,
+        }
+        .into()
+    }
+
+    pub fn name(&self) -> String {
+        match self {
+            SupportProtocols::Ping => "/ckb/ping",
+            SupportProtocols::Discovery => "/ckb/discovery",
+            SupportProtocols::Identify => "/ckb/identify",
+            SupportProtocols::Feeler => "/ckb/flr",
+            SupportProtocols::DisconnectMessage => "/ckb/disconnectmsg",
+            SupportProtocols::Sync => "/ckb/syn",
+            SupportProtocols::Relay => "/ckb/rel",
+            SupportProtocols::Time => "/ckb/tim",
+            SupportProtocols::Alert => "/ckb/alt",
+        }
+        .to_owned()
+    }
+
+    pub fn support_versions(&self) -> Vec<String> {
```

Comment:
Why there are different versions, `"0.0.1"` and `"1"`?

Reply:
we didn't invoke `MetaBuilder#support_versions` fn for these protocols (ping/discovery/etc) in previous code, so the default `0.0.1` value is used:
https://github.com/nervosnetwork/tentacle/blob/master/src/builder.rs#L312

have to keep `0.0.1` for compatibility...

Reply:
Please add some comments to describe this history.

Reply:
copied my comment to code, please review again, thanks.